### PR TITLE
Add CI for MacOS

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
This adds MacOS-based CI with minimal changes to the workflow document.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml:17]: Changes are probable best viewed with "Hide whitespace", since this introduces an `if` which changes existing script indentation.

File: .github/workflows/ci.yml
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml:25]: The only differences, from Ubuntu, are this section and the one below (`KERL_RELEASE_TARGET`). I use Homebrew to ease the installation process of required tools...

File: .github/workflows/ci.yml
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml:40]: I single this `if` out so it's easier to check for the differences (and hopefully one day they can be the same - btw, we're not doing target `asan` for Ubuntu).

The targets were not testing aren't straightforward to achieve, at least with Homebrew. e.g. `valgrind` would come from [an alternative source](https://github.com/LouisBrunner/homebrew-valgrind), `gcov` and `gprof` supposedly comes from [Command Line Tools](https://mac.install.guide/commandlinetools/index.html), but I can't seem to find `gcov` after installing those, and `gprof`'s been deprecated, ...

If all goes well this can still be an acceptable first base for building upon.

**Review by jadeallenx — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
@jadeallenx, ready to merge when you are 😄

**Comment by jadeallenx** [comment]
Thanks

**Comment by jadeallenx** [comment]
At this point I want to give you a commit bit for this repo if you want one...

File: .github/workflows/ci.yml
Comment (garazdawi) [reply .github/workflows/ci.yml:40]: If `debug` and `opt` work then the rest should also work. No need to make things extra complicated in macOS.

Comment (paulo-ferraz-oliveira) [reply .github/workflows/ci.yml:40]: Hey, @garazdawi. I tried it with `debug opt gcov`, for example, and stuff started failing (thus the `if`). I didn't want to dig further, since I believed this was Ok for a first step, but maybe I could open a pull request where I add just `gcov` (so you can see the issue) and maybe have your 👀.

**Comment by paulo-ferraz-oliveira** [comment]
Hey, @jadeallenx. What's a commit bit? I don't think I've come across the term. Is that like collaboration rights? I'd be Ok with that, sure. Thank you.

**Comment by jadeallenx** [comment]
> Hey, @jadeallenx. What's a commit bit? I don't think I've come across the term. Is that like collaboration rights? I'd be Ok with that, sure. Thank you.

It's permission to make commits to this repository

**Pull request merged**
